pkg/event/registry.go

```diff
@@ -63,6 +63,12 @@ func (er *Registry) AddHandlers(eventHandlers ...Handler) error {
 	return nil
 }
 
+func (er *Registry) StartHandlers(clusterID string, ClusterCidr, ServiceCidr []string) error {
```

Comment:
I'm doing this in the OVN side when receiving the first local-endpoint.

May be we don't need to receive this anymore for the route agents and we can discover it from the local endpoints?

The controller already knows the local clusterID... and will send LocalEndpoint... updates

Reply:
I thought about it, but its prone to race conditions if we take that approach. Let me explain...

As you know, route-agent needs to discover the local CNI Interface on the hosts and also annotate the nodes with the cni-iface ip. As part of discoverCNIInterface API, we need the localClusterCIDR info. I plan to handle this in the NodeCreated event. So, when NodeCreated event is fired, we need the localClusterCIDR.
If we use the localEndpoint events, it might arrive later (most likely) compared to NodeEvents.
So to have a proper initialization sequence, IMHO its better to have a new API called "Start" (or something on those lines) where we can initialize the necessary things on the handler side. Hope it makes sense.

Reply:
Makes sense, it's not a problem for OVN.

Can you take those details from the environment variables instead in the driver itself?


Also, can we separate the "annotation of local CNI ip" to a separate handler?,

In that way it would be reusable if we wanted to use it also... in OVN or other drivers for example.

Also code is cleaner and more isolated.

Reply:
Yes, the controller.go is getting the info from the environment variables and is passing it to the handlers via StartHandlers() API.

You mean an handler to "annotate the nodes with CNI IP"? something like logger handler?

Reply:
Yes, something like that, if that makes sense.

That way it would be a reusable functionality for OVN too.

Reply:
> Yes, the controller.go is getting the info from the environment variables and is passing it to the handlers via StartHandlers() API.
> 
> You mean an handler to "annotate the nodes with CNI IP"? something like logger handler?

I mean, instead of doing that, if you need that info, grab it from the environment variables. Could also be passed during Init, why having a separate start handlers?